codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/CoreCodegenContext.kt

```diff
@@ -3,12 +3,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package software.amazon.smithy.rust.codegen.client.smithy
+package software.amazon.smithy.rust.codegen.core.smithy
```

Comment:
I think we should rename the `Core-` prefix out of these now that they're in the actual core module.

Reply:
Agree. Done in [Rename CoreCodegenContext as CodegenContext, since it's now in th…](https://github.com/awslabs/smithy-rs/pull/1762/commits/90d250dbb5ef6781050f623c14d58e124d819d28).